x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -86,14 +85,16 @@ func parseSingle(b []byte) (statsdMetric, error) {
 	return s, nil
 }
 
-// parse will parse a statsd metric into its components
+// parse will parse statsd metrics into individual metric and then its components
 func parse(b []byte) ([]statsdMetric, error) {
-	metrics := []statsdMetric{}
-	for _, rawMetric := range bytes.Split(b, []byte("\n")) {
-		if len(rawMetric) > 0 {
-			metric, err := parseSingle(rawMetric)
+	rawMetrics := bytes.Split(b, []byte("\n"))
+	metrics := make([]statsdMetric, 0, len(rawMetrics))
+	for i := range rawMetrics {
+		if len(rawMetrics[i]) > 0 {
+			metric, err := parseSingle(rawMetrics[i])
 			if err != nil {
-				return metrics, err
+				logger.Warnf("invalid packet: %s", err)
```

Comment:
Should this be an error instead?

Reply:
I even spent some thinking about it when I was making the change. But in my opinion, `Warn` is more suitable than `Error` because the client is expected to send packets that conform with the basic line protocol that a statsd metrics parser expects. But, a buggy client can also send packets with the non-standard format and we should skip them and not let it affect our processing.

If I refer to definitions of `Warn` and `Error` from [here](https://sematext.com/blog/logging-levels/):

> **Warn:** Unexpected behaviour happened inside the application, but it is continuing its work and the key business features are operating as expected.
>
> **Error:** One or more functionalities are not working, preventing some functionalities from working correctly.

So, `Warn` seems to be a better choice because something did unexpectedly happen inside the application but it didn't affect any feature and it is even continuing to work.

What do you think?